File: src/main/scala/committee/nova/skillful/event/impl/SkillXpEvent.scala
package committee.nova.skillful.event.impl

import committee.nova.skillful.skills.SkillInstance
import net.minecraft.entity.player.EntityPlayerMP

object SkillXpEvent {
  class Pre(private val player: EntityPlayerMP, private val skill: SkillInstance, private var amount: Int) extends SkillXpEvent(player, skill, amount) {
    def setAmount(newAmount: Int): Unit = amount = newAmount

    override def isCancelable: Boolean = true
  }

  class Post(private val player: EntityPlayerMP, private val skill: SkillInstance, private var amount: Int) extends SkillXpEvent(player, skill, amount)
}

class SkillXpEvent(private val player: EntityPlayerMP, private val skill: SkillInstance, private var amount: Int) extends SkillEvent(player, skill) {
  def getAmount: Int = amount
}

File: src/main/java/committee/nova/skillful/common/event/impl/SkillXpEvent.java
package committee.nova.skillful.common.event.impl;

import committee.nova.skillful.common.skill.SkillInstance;
import net.minecraft.world.entity.player.Player;

public class SkillXpEvent extends SkillEvent {
    private final long change;
    private final long lvlChange;

    public SkillXpEvent(Player player, SkillInstance skill, long change, long lvlChange) {
        super(player, skill);
        this.change = change;
        this.lvlChange = lvlChange;
    }

    public long getChange() {
        return change;
    }

    public long getLvlChange() {
        return lvlChange;
    }

    public boolean isUp() {
        return lvlChange > 0 || (lvlChange == 0 && change > 0);
    }

    public boolean isDown() {
        return lvlChange < 0 || (lvlChange == 0 && change < 0);
    }
}
